ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,107 @@
+# 4. Out-of-the-box image repository for AppStudio users 
+
+Date: Oct 29, 2022 
+
+## Status
+
+Accepted
+
+## Approvers
+
+* Alexey Kazakov
+* Gorkem Ercan
+
+## Reviewers
+
+* Alexey Kazakov
+* Gorkem Ercan
+* Parag Dave
+
+
+## Context
+
+**Problem**
+AppStudio does not have a internal registry where images could be pushed to as an intermediate step before being deployed as a container.
+As an application centric experience, AppStudio should not make it mandatory for its users to specify where the image ( built from source code ) needs
+to be pushed to.
+
+
+**Goals**: 
+* Provide an out-of-the-box location for users’ images to be pushed after being built from source.
+
+
+**Non-Goals**: 
+* Define the user experience for users bringing existing images from other image registry services.
+* Provide users an option to choose what the out-of-the-box location for images would be. 
+* Define the user experience for users who are willing to configure additional credentials for pushing to an image registry of their choice.
+
+**Design Goals**
+* Use Quay.io as the out-of-the-box image registry in order to avoid getting into the business of maintaining an internal registry.
+* Align user permissions in an AppStudio workspace with those in Quay.io
+* Maintain the right levels of isolation between images being expected of a multi-tenant system.
+* Security:
+  * Leverage short-lived Quay.io API tokens for repo/org management on Quay.io
+  * Leverage robot accounts for pushing images from CI/Build pipelines.
+
+
+
+## Decision
+
+### What
+
```

Comment:
Can we also discuss here what happens to the repo/images when the workspace or user is deleted?